### PR TITLE
docs(github): Update github commit guidelines and remove duplicate contribution guidelines doc

### PR DESCRIPTION
**Pull request by vinagesh**
## Description of the changes
Updated C# SDK commit guidelines so that it is easy to write release notes.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
...
 **If your contribution is going to be a major effort, you should give us a heads-up first. We have a lot of items captured in our backlog and we release every two weeks, so before you spend the time, just check with us to make sure your plans and ours are in sync :) Just open an issue on GitHub and tag it as "contribution".**
 
+#### Commit Guidelines
+We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.
+
+**Commit Message Format**
+
+Each commit message consists of a header, a body (optional) and a footer (optional). The header has a special format that includes a type, a scope and a subject:
+
+```java
```
Comment (barustum) [inline .github/CONTRIBUTING.md:85]: C#

Comment (vinagesh) [reply .github/CONTRIBUTING.md]: Fixed just a second before you posted :)

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 * Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
 * Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
 * Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
-* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).
+* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C#, C, Java, .NET, Node.js, Python).
```
Comment (abhipsaMisra) [inline .github/CONTRIBUTING.md:10]: Add a note that this repository is for the C# SDK only. Issues found for other languages should be filed under their respective repositories.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
 
 **If your contribution is going to be a major effort, you should give us a heads-up first. We have a lot of items captured in our backlog and we release every two weeks, so before you spend the time, just check with us to make sure your plans and ours are in sync :) Just open an issue on GitHub and tag it as "contribution".**
 
+#### Commit Guidelines
```
Comment (abhipsaMisra) [inline .github/CONTRIBUTING.md:78]: Question - Why do we have 2 different md files for contributing guidelines? - `./CONTRIBUTING.md` and `./.github/CONTRIBUTING.md`
We should remove the one that is not required.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
...
+```
+The header is mandatory and the body and footer are optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.
+
+Footer should contain a [closing reference](#https://help.github.com/articles/closing-issues-using-keywords/) to an issue if any.
+
+**Revert**
+
+If the commit reverts a previous commit, it should begin with revert:, followed by the header of the reverted commit. In the body it should say: This reverts commit <hash>., where the hash is the SHA of the commit being reverted.
```
Comment (abhipsaMisra) [inline .github/CONTRIBUTING.md:102]: nit: mark the keyword and sample working as code (`revert:` etc)

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
...
+
+* style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* test: Adding missing tests or correcting existing tests
+
+**Scope**
+
+The scope should be the name of the package affected as perceived by person reading changelog generated from commit messages.
+
+    * shared
+    * devices-client
```
Comment (abhipsaMisra) [inline .github/CONTRIBUTING.md:130]: To maintain parity with dps, this should be called `iothub-client`, `iothub-service`.

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
...
+    * service-client
+    * provisioning-client
+    * provisioning-service
+    * transport-amqp
```
Comment (abhipsaMisra) [inline .github/CONTRIBUTING.md:134]: these are provisioning specific transport libraries - `provisioning-transport-amqp` etc

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (vinagesh) [reply .github/CONTRIBUTING.md:78]: Good observation :) Let me see which one we want to keep.

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (vinagesh) [reply .github/CONTRIBUTING.md:102]: I did not get you.

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (vinagesh) [reply .github/CONTRIBUTING.md:130]: Sounds good!

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (vinagesh) [reply .github/CONTRIBUTING.md:134]: Sounds good!

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (abhipsaMisra) [reply .github/CONTRIBUTING.md:102]: I meant, something like this:
If the commit reverts a previous commit, it should begin with `revert:`, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -7,7 +7,7 @@
 * Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
 * Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
 * Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
-* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).
+* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub. This is the repo for C# only. Use the respective repo for (C, Java, .NET, Node.js, Python).
```
Comment (prmathur-microsoft) [inline .github/CONTRIBUTING.md:11]: Link to respective repo?

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
...
+    * iothub-client
+    * iothub-service
+    * provisioning-client
+    * provisioning-service
+    * provisioning-transport-amqp
+    * provisioning-transport-mqtt
+    * provisioning-transport-http
+    * provisioning-security-tpm
+    * e2e
+    * github
```
Comment (prmathur-microsoft) [inline .github/CONTRIBUTING.md:139]: Can we add pnp-device, pnp-service to this list

**Pull request merged**